clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -377,5 +377,15 @@ public void moveToNextAddress() throws UnknownHostException {
         public String toString() {
             return "NodeState(" + state + ", " + lastConnectAttemptMs + ", " + failedAttempts + ", " + throttleUntilTimeMs + ")";
         }
+
+        /* Visible for testing */
+        private void setHost(String host) {
+            this.host = host;
+        }
+    }
+
+    /* Visible for testing */
+    protected void changeHostForNode(String nodeId, String host) {
```

Comment:
visibility could be restricted to be package-level (i.e remove keyword `protected`)

Reply:
Perhaps it would be better to change host using reflection in the test since this code looks out of place in the implementation? Then the `host` field can also be made final.